attribute-service/src/main/java/org/hypertrace/core/attribute/service/AttributeServiceEntry.java

```diff
@@ -48,11 +53,14 @@ protected void doStart() {
   }
 
   @Override
-  protected void doStop() {}
+  protected void doStop() {
+    healthStatusManager.enterTerminalState();
+    ServerManagementUtil.shutdownServer(this.server, this.getServiceName(), Duration.ofMinutes(1));
+  }
 
   @Override
   public boolean healthCheck() {
-    return true;
+    throw new UnsupportedOperationException("Please use grpc health check");
```

Comment:
Alternatively, I can try to serve this one up, too. Thee issue is that health status manager only exposes write apis, I see no way of reading the value without a full grpc client.

Reply:
ah, integration tests use it. thought this might backfire. So I can make that path work, or I can try to pipe the grpc result into here...

Reply:
> pipe the grpc result into here.

This would be ideal in my opinion